### DISCUSSION TITLE
Lesson 6 brownie test

### DISCUSSION THREAD

**Discussion by avikat** [smartcontractkit/full-blockchain-solidity-course-py#961] [Q&A] [answered]
Getting this error for brownie test
    

```
def test_can_fund_withdraw():
        account = get_account()
        fund_me = deply_fund_me()
>       entrance_fee = fund_me.getEntranceFee()
E       AttributeError: 'NoneType' object has no attribute 'getEntranceFee'

tests\test_fund_me.py:8: AttributeError
------------------------------ Captured stdout call -------------------------------
The network is development
Deploying Mocks ....
Mocks Deployed
Contract deploy to 0x602C71e4DAC47a042Ee7f46E0aee17F94A3bA0B6
============================= short test summary info =============================
FAILED tests/test_fund_me.py::test_can_fund_withdraw - AttributeError: 'NoneType'...=============================== 1 failed in 10.24s ================================
Terminating local RPC client...
```

**Comment by cromewar** [comment]
Hello @avikat 
You have a typo here:
fund_me = deply_fund_me()
the correct one is deploy, isn't it?

**Reply by avikat** [reply]
Hiii Actually i have deply_fund_me function only everywhere. and deploy.py is running fine

**Reply by cromewar** [reply]
I've checked your code on another post, seems ok.
Could you please run the test as follows:

 `brownie test -s -k test_can_fund_withdraw`

And share the error log here please.
Also please do the same @Zeyad20080

**Comment by Zeyad20080** [comment]
Hey there I get also the same error

**Comment by gnoparus** [comment, marked as answer]
I've got the same problem. 

`AttributeError:` 'NoneType' object has no attribute '_with_attr'


After I did some research in this discussion forum, I tried reinstall ganache-cli as follows.

uninstall ganache-cli
```
npm uninstall ganache-cli
sudo npm uninstall ganache-cli --global

```
reinstall ganache-cli
```
sudo npm install ganache-cli --global
npm install ganache-cli

```
test again
`brownie test -k test_only_owner_can_withdraw`

And it worked.

`FAILED tests/test_fund_me.py::test_only_owner_can_withdraw - brownie.exceptions.VirtualMachineError: revert`

There are node_modules/ and package-lock.json appear on brownie_fund_me project path.

**Reply by gnoparus** [reply]
Reference:
https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/295#discussioncomment-1599106

**Reply by gnoparus** [reply]
My environment is windows 10 / WSL2 - Ubuntu 20.04 / Python 3.8.10

**Reply by cromewar** [reply]
Great job @gnoparus !!

**Comment by avikat** [comment]
Thank you so much. There are best people here in this Community. I will also try to help others
